test/common/conn_pool/conn_pool_base_test.cc

```diff
@@ -374,13 +374,13 @@ TEST_F(ConnPoolImplDispatcherBaseTest, MaxConnectionDurationBusy) {
   // Verify that advancing to just before the connection duration timeout doesn't drain the
   // connection.
   advanceTimeAndRun(max_connection_duration_ - 1);
-  EXPECT_EQ(0, pool_.host()->cluster().trafficStats().upstream_cx_max_duration_reached_.value());
+  EXPECT_EQ(0, pool_.host()->cluster().trafficStats()->upstream_cx_max_duration_reached_.value());
```

Comment:
optional put the instantiated traffic stats in a temp. Or maybe even in member variable, just to make the code less cluttered.